src/domain/mongoose_domain_loader.erl

```diff
@@ -27,44 +28,50 @@ load_data_from_base_loop(FromId, PageSize) ->
             load_data_from_base_loop(PageMaxId, PageSize)
     end.
 
+remove_outdated_domains_from_core() ->
+    OutdatedDomains = mongoose_domain_core:get_all_outdated(),
+    remove_domains(OutdatedDomains).
+
 check_for_updates(FromId, PageSize) ->
-    check_if_from_num_still_relevant(FromId),
-    %% Ordered by the earlist events first
+    %% Ordered by the earliest events first
     try mongoose_domain_sql:select_updates_from(FromId, PageSize) of
-        [] -> FromId;
         Rows ->
-            PageMaxId = row_to_id(lists:last(Rows)),
-            apply_changes(Rows),
-            check_for_updates(PageMaxId, PageSize)
+            case check_if_id_is_still_relevant(FromId, Rows) of
```

Comment:
Why check it while paging? It's not possible to lose the connection to the DB for a long time while paging...

Reply:
you never know when exactly another node can run table clean up

Reply:
Ok, let's keep it then.